### ISSUE TITLE
Impact of training data size unknown

### ISSUE THREAD

**Issue by tom9358** [tom9358/nllb-tryout#1]
In tandem with messing around with the code, I regularly write Gronings sentences with translations and add them to https://tatoeba.org/. In addition to wanting to expand the publicly available data on Gronings, I do this because the current corpus of parallel sentences seemed to be insufficient. However, I don't really know yet how the amount of training data translates (pun intended) to machine translation quality. I'm also not sure what the limits of (different sizes of) the NLLB models are. Time to investigate this all in a bit more detail!

I'll train different sizes of NLLB models with different amounts of training data, and I'll add German Low Saxon / Low German [nds] to my models, since it has a bit more training data than Gronings Low Saxon [gos] and it is a structurally very similar language, so I would expect results to be quite comparable between the two. As an additional path of investigation, I'll train models on both languages simultaneously and see if some kind of cross-pollination happens. I'll use (an) automated metric(s) such as BLEU, ChrF, an edit distance like the Levenshtein distance to guesstimate the translation quality. Maybe I'll also do some (blind) human evaluation, I already have some code for that too - we'll see.

To do:
- [ ] add nds to tokenizer and model
- [ ] add code to load nds data
- [x] create more structured way to save models and parameters
- [x] create way to save metric scores
- [ ] train `facebook/nllb-200-distilled-600M` and `facebook/nllb-200-distilled-1.3B` on at least 2500, 5000, 10000, 15000, 18000 (the total for Low Saxon data), calculate scores and plot them. Do so for different amounts of training steps and batch sizes.
  - [ ] only distilled 1.3B model, nds - deu data, for something like 500 steps with batch size 16 (roughly, adapt if it seems to be over- or underfitting)
  - [ ] try the same for the 600M model
  - [ ] add nds - eng data and see if the German scores improve!
  - [ ] add gos data!
  - [ ] I'll add more todo boxes here later. 
- [ ] look for other languages I could do this with (i.e. which aren't already present in the NLLB models)

**Comment by tom9358** [comment]
Well, from discussing with VanModers I understand that 10000 pairs is a rather small. I haven't tried this entire plan and it doesn't answer the question of whether using more _multilingual_ training data is a good idea, but for now I find it more interesting to just accept my amount of training data (although I'm interested in expanding with e.g. dictionary data or synthetic sentences), and looking at some hyperparameter tuning. I'm sure I've overtrained the latest two models, actually.

**Comment by tom9358** [comment]
I haven't looked at tatoeba nds data, but I do know that both multilingual training and using a 1.3B model instead of a 600M one both improve performance. My current repo does not support adding two languages yet, but this would mean that a single model then has two separate language's performance to improve on. It makes much more sense to just train two models - unless the linguistic transfer would be huge. There is no reason to believe transfer would be higher than with Dutch/German/English, especially considering that those latter three are not low-resource languages. tldr; closing as too broad and doesn't make sense anymore after some voortschrijdend inzicht
